tests/models/abiotic/test_abiotic_tools.py

```diff
@@ -31,24 +82,191 @@ def test_calc_molar_density_air(dummy_data):
 
     from virtual_rainforest.models.abiotic import abiotic_tools
 
+    data = dummy_data
+    layers_function = abiotic_tools.set_layers_function(canopy_layers=3, soil_layers=1)
+
     result = abiotic_tools.calc_molar_density_air(
-        temperature=dummy_data.data["temperature_2m"],
-        atmospheric_pressure=dummy_data.data["atmospheric_pressure"],
+        temperature=data["temperature_2m"],
+        atmospheric_pressure=data["atmospheric_pressure"],
     )
     xr.testing.assert_allclose(
-        result, DataArray([120.618713, 119.436176, 118.276602], dims=["cell_id"])
+        result,
+        DataArray(
+            [[120.618713, 119.436176, 118.276602]],
+            dims=["layers", "cell_id"],
+            coords={
+                "layers": [len(layers_function) - 1],
+                "layers_function": ("layers", ["above"]),
+                "cell_id": [0, 1, 2],
+            },
+            name="molar_density_air",
+        ),
     )
 
 
 def test_calc_specific_heat_air(dummy_data):
-    """Test canclualte specific heat of air."""
+    """Test calculate specific heat of air."""
+
+    from virtual_rainforest.models.abiotic import abiotic_tools
+
+    data = dummy_data
+    layers_function = abiotic_tools.set_layers_function(canopy_layers=3, soil_layers=1)
+
+    result = abiotic_tools.calc_specific_heat_air(temperature=data["temperature_2m"])
```

Comment:
All the test functions have the same lines to create `data` and `layers_function`. I'd make `layers_function` a fixture and just use `dummy_data` directly.

```python
def test_calc_specific_heat_air(dummy_data, layers_function):
    """Test calculate specific heat of air."""

    from virtual_rainforest.models.abiotic import abiotic_tools

    result = abiotic_tools.calc_specific_heat_air(temperature=dummy_data["temperature_2m"])
```

Reply:
I hard coded the layers that are expected, so this is outdated. For other scripts I will definitely follow this suggestion.